src/scripts/qa.ts

```diff
@@ -59,633 +46,40 @@ async function getAuthKey(): Promise<{ name: string; key: string }> {
   }
 }
 
-async function fetchQuoteChains(
-  hashflow: HashflowApi,
-  chain: Chain
-): Promise<Chain[]> {
-  process.stdout.write(`Fetching all available quote chains...`);
-  const tradingPairs = await hashflow.getTradingPairs(chain);
-  const chains = tradingPairs.pairs.map(p => p.quoteToken.chain);
-  const seen: string[] = [];
-  const dedupChains: Chain[] = [];
-  chains.forEach(chain => {
-    if (!seen.includes(stringifyChain(chain))) {
-      dedupChains.push(chain);
-      seen.push(stringifyChain(chain));
-    }
-  });
-  return dedupChains;
-}
-
-function stringifyChain(chain: Chain): string {
-  return `${chain.chainType}_${chain.chainId}`;
-}
-
-async function handler(): Promise<void> {
+async function main() {
   const { name, key } = await getAuthKey();
 
   const argv = await parser.argv;
-
   const maker = argv.maker;
   const chainType = argv.chain_type as ChainType;
   const chainId = argv.chain as ChainId;
   const chain: Chain = { chainType, chainId };
   const env = argv.env as Environment;
-  validateMakerName(maker);
-  validateChain(chain);
-  validateEnvironment(env);
-
-  const crossChainCheck = argv.check_all_xchain;
-
-  if (crossChainCheck && argv.quote_chain) {
-    throw new Error(
-      `Can't specify a quote_chain and check_all_xchain simultaneously`
-    );
-  }
-
-  const maybeQuoteChain = argv.quote_chain
-    ? {
-        chainType: argv.quote_chain_type as ChainType,
-        chainId: argv.quote_chain as ChainId,
-      }
+  const quoteChainType = argv.quote_chain_type as ChainType;
+  const quoteChainId = argv.quote_chain as ChainId;
+  const quoteChain = quoteChainId
+    ? { chainId: quoteChainId, chainType: quoteChainType }
```

Comment:
Not necessarily for this change but I was reading about type casting vs narrowing in typescript and generally we should be using type narrowing rather than casting to give us better type safety.

https://byby.dev/ts-type-narrowing